subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/AttributeConfigurationSelector.java

```diff
@@ -137,16 +151,18 @@ private static List<VariantGraphResolveMetadata> findVariantsProvidingExactlySam
         return sameClassifier;
     }
 
-    private static VariantSelectionResult singleVariant(Optional<List<? extends VariantGraphResolveMetadata>> variantsForGraphTraversal, List<VariantGraphResolveMetadata> matches) {
-        return new VariantSelectionResult(ImmutableList.of(matches.get(0)), variantsForGraphTraversal.isPresent());
+    private static VariantSelectionResult singleVariant(boolean variantAware, List<VariantGraphResolveMetadata> matches) {
+        assert matches.size() == 1;
```

Comment:
🧐 Assertion in prod code might not run.

Reply:
This is more like documentation that happens to be verified during tests. Given it's a private utility method with an otherwise obvious method name and clear purpose, I would say a full `Preconditions` check here with an error message is overkill.